### PR TITLE
Added make_hgrid C files for wrapping

### PR DESCRIPTION
**Pull request by fmalatino**
**Description**
This PR adds the C files to be wrapped and called by `make_hgrid.py`

**Tested**
No unit tests exist for this as of yet. As wrappers are generated, unit tests will be developed to ensure their functionality.

### CODE REVIEW

File: gridtools_lib/clib/make_hgrid/empty.py [hunk 1]
```diff
@@ -1 +0,0 @@
-# For git tracking, to be removed
```
Comment (mlee03) [inline gridtools_lib/clib/make_hgrid/empty.py:1]: is this file needed?

**Review by mlee03 — CHANGES_REQUESTED** [review]

File: gridtools_lib/clib/make_hgrid/empty.py (hunk 1 above)
Comment (fmalatino) [reply gridtools_lib/clib/make_hgrid/empty.py:1]: No it is not, that is why I removed it.

**Review by mlee03 — APPROVED** [review]
